lib/Email/Stuffer/TestLinks.pm

```diff
@@ -0,0 +1,77 @@
+package Email::Stuffer::TestLinks;
+
+use strict;
+use warnings;
+
+our $VERSION = '0.010';
+
+use Test::Most;
+use Mojo::UserAgent;
+use Email::Stuffer;
+use Class::Method::Modifiers qw/ install_modifier /;
+
+=head1 SYNOPSIS
+
+    use Email::Stuffer::TestLinks;
+
+=head1 NAME
+
+Email::Stuffer::TestLinks - validates links in HTML emails sent by Email::Stuffer>send_or_die()
+
+=head1 DESCRIPTION
+
+When this module is included in a test, it parse HTML links in every email sent through check Email::Stuffer.
+Only html links (<a href="xyz"...) are checked. Each link must be valid resource and the returned page title must not contain 'error' or 'not found'.
+
+=cut
+
+install_modifier 'Email::Stuffer', after => send_or_die => sub {
+
+    my $self = shift;
+    my $ua   = Mojo::UserAgent->new;
+    $ua->max_redirects(10);
+    $ua->connect_timeout(5);
+
+    my @urls;
+    my $body;
+    $self->email->walk_parts(
+        sub {
+            my ($part) = @_;
+            if ( $part->content_type && $part->content_type =~ /text\/html/i ) {
+                my $dom = Mojo::DOM->new( $part->body );
+                my $links = $dom->find('a')->map( attr => 'href' )->compact;
+
+                # Exclude anchors, mailto
+                push( @urls, grep { !/^(#|mailto)/ } @$links );
```

Comment:
Might as well just whitelist all `/^https?:/` URLs here, and since these are emails I'd have a test that flags anything without a scheme (e.g. `/xxx` or `#abc`).

Reply:
URLs like "binary.com" should and do work, so what I've done is exclude mailto:, and then add a test to fail things that are obviously anchors or relative links since they never belong in emails.